File: packages/create-react-native-library/templates/java-library-mixed/android/src/oldarch/{%- project.name %}Spec.java
package com.<%- project.package %>;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Promise;

abstract class <%- project.name -%>Spec extends ReactContextBaseJavaModule {
  <%- project.name -%>Spec(ReactApplicationContext context) {
    super(context);
  }

  public abstract void multiply(double a, double b, Promise promise);
}

File: packages/create-react-native-library/templates/kotlin-view-library-mixed/android/src/oldarch/{%- project.name %}Spec.kt
package com.<%- project.package %>

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.Promise

abstract class <%- project.name -%>Spec internal constructor(context: ReactApplicationContext) :
  ReactContextBaseJavaModule(context) {

  abstract fun multiply(a: Double, b: Double, promise: Promise)
}
